src/ansiblelint/utils.py

```diff
@@ -759,6 +761,60 @@ def construct_mapping(
     return data
 
 
+def iter_tasks_in_file(
+    lintable: Lintable, rule_id: str
+) -> Iterator[Tuple[Dict[str, Any], Dict[str, Any], bool, Optional[MatchError]]]:
+    """Iterate over tasks in file.
+
+    This yields a 4-tuple of raw_task, normalized_task, skipped, and error.
+
+    raw_task:
+        When looping through the tasks in the file, each "raw_task" is minimally
+        processed to include these special keys: __line__, __file__, skipped_rules.
+    normalized_task:
+        When each raw_task is "normalized", action shorthand (strings) get parsed
+        by ansible into python objects and the action key gets normalized. If the task
+        should be skipped (skipped is True) or normalizing it fails (error is not None)
+        then this is just the raw_task instead of a normalized copy.
+    skipped:
+        Whether or not the task should be skipped according to tags or skipped_rules.
+    error:
+        This is normally None. It will be a MatchError when the raw_task cannot be
+        normalized due to an AnsibleParserError.
+
+    :param lintable: The playbook or tasks/handlers yaml file to get tasks from
+    :param rule_id: The current rule id to allow calculating skipped.
+
+    :return: raw_task, normalized_task, skipped, error
+    """
+    data = parse_yaml_linenumbers(lintable)
+    if not data:
+        return
+    data = ansiblelint.skip_utils.append_skipped_rules(data, lintable)
+
+    raw_tasks = get_action_tasks(data, lintable)
+
+    for raw_task in raw_tasks:
+        # An empty `tags` block causes `None` to be returned if
+        # the `or []` is not present - `task.get('tags', [])`
+        # does not suffice.
+        if 'skip_ansible_lint' in (raw_task.get('tags') or []) or (
+            rule_id in raw_task.get('skipped_rules', ())
+        ):
```

Comment:
Look how we can add context via "label-variables":
```suggestion
        skip_with_a_yaml_comment = 'skip_ansible_lint' in (raw_task.get('tags') or [])
        skip_with_a_task_tag = rule_id in raw_task.get('skipped_rules', ())
        if skip_with_a_yaml_comment or skip_with_a_task_tag:
```

Reply:
yeah... pylint+flake8 has taught me to avoid creating new vars wherever possible to keep the complexity low. But I can add some better vars here.

Reply:
done